src/fvm_multicell.hpp

```diff
@@ -518,12 +529,39 @@ void fvm_multicell<Backend>::initialize(
             map_entry.push_back(syn_cv);
         }
 
+        //
         // add the stimuli
+        //
+
+        // step 1: pack the index and parameter information into flat vectors
+        std::vector<size_type> stim_index;
+        std::vector<value_type> stim_durations;
+        std::vector<value_type> stim_delays;
+        std::vector<value_type> stim_amplitudes;
         for (const auto& stim: c.stimuli()) {
             auto idx = comp_ival.first+find_cv_index(stim.location, graph);
-            stimuli_.push_back({idx, stim.clamp});
+            stim_index.push_back(idx);
+            stim_durations.push_back(stim.clamp.duration());
+            stim_delays.push_back(stim.clamp.delay());
+            stim_amplitudes.push_back(stim.clamp.amplitude());
         }
 
+        // step 2: create the stimulus mechanism and initialize the stimulus
+        //         parameters
+        // NOTE: the indexes and associated metadata (durations, delays,
+        //       amplitudes) have not been permuted to ascending cv index order,
+        //       as is the case with other point processes.
+        //       This is because the hard-coded stimulus mechanism makes no
+        //       optimizations that rely on this assumption (they are not
+        //       performance critical).
+        if (stim_index.size()) {
+            auto stim = new stimulus(
+                voltage_, current_, memory::make_const_view(stim_index));
```

Comment:
I don't think there is a guarantee that the `stim_index` will be sorted. Perhaps it would be better to generalize the `syn_mech_indices` machinery? The problem of tracking targets and making target handles is also analogous to the problem of tracking entries in the `c.stimuli()` and assigning the appropriate stimulus parameters.